src/cpp/session/modules/SessionFind.cpp

```diff
@@ -608,10 +1195,76 @@ core::Error initialize()
    initBlock.addFunctions()
       (bind(registerRpcMethod, "begin_find", beginFind))
       (bind(registerRpcMethod, "stop_find", stopFind))
-      (bind(registerRpcMethod, "clear_find_results", clearFindResults));
+      (bind(registerRpcMethod, "clear_find_results", clearFindResults))
+      (bind(registerRpcMethod, "preview_replace", previewReplace))
+      (bind(registerRpcMethod, "complete_replace", completeReplace))
+      (bind(registerRpcMethod, "stop_replace", stopReplace));
    return initBlock.execute();
 }
 
+core::Error Replacer::completeReplace(const boost::regex* searchRegex,
+                                      const std::string* replaceRegex,
+                                      size_t matchOn, size_t matchOff, std::string* line,
+                                      size_t* pReplaceMatchOff)
+{
+   std::string temp;
+   try
+   {
+      temp = boost::regex_replace(*line, *searchRegex, *replaceRegex, boost::format_sed);
+   }
+   catch (const std::runtime_error& e)
+   {
+      return core::Error(-1, e.what(), ERROR_LOCATION);
```

Comment:
I think that an actual error code of some kind is needed here (a generic code is fine). @MariaSemple just refactored all the error codes so can probably offer better guidance.